exec/cnex/cnex.c

```diff
@@ -28,28 +30,88 @@ typedef struct tagTCommandLineOptions {
     int  ArgStart;
     char *pszFilename;
     char *pszExecutableName;
+    char *pszExecutablePath;
 } TOptions;
 
-static TOptions gOptions = { FALSE, FALSE, TRUE, 0, NULL, NULL };
+static TOptions gOptions = { FALSE, FALSE, TRUE, 0, NULL, NULL, NULL };
 
-void exec_freeExecutor(TExecutor *e)
+static char *getApplicationName(char *arg)
 {
-    uint32_t i;
+    char *p = arg;
+    char *s = p;
+
+    while (*p) {
+        if (*p == '\\' || *p == '/') {
+            s = p;
+        }
+        p++;
+    }
+    s++;
+    return s;
+}
 
-    for (i = 0; i < e->object->global_size; i++) {
-        cell_clearCell(&e->globals[i]);
+static char *getApplicationPath(char *arg)
+{
+    BOOL bHavePath = TRUE;
+    char *p = arg;  // p pointer to provided arg.
+    char *e = p;    // e Last path seperator found.
+
+    // Walk the arg string, recording the last position of a path seperator char.
+    while (*p) {
+        if (*p == '\\' || *p == '/' || *p == ':') {
```

Comment:
Why is this different from the implementation in `getApplicationName`? On some platforms, a colon is a perfectly valid part of a file name, too.